code/modules/modular_computers/laptop_vendor.dm

```diff
@@ -288,35 +288,52 @@ obj/machinery/lapvend/attackby(obj/item/weapon/W as obj, mob/user as mob)
 
 
 // Simplified payment processing, returns 1 on success.
-/obj/machinery/lapvend/proc/process_payment(var/obj/item/weapon/card/id/I, var/obj/item/ID_container)
+/obj/machinery/lapvend/proc/process_payment(var/obj/item/weapon/card/id/I, var/obj/item/ID_container, var/obj/item/weapon/spacecash/S)
```

Comment:
This won't work. `process_payment()` is called with `I, W` in the same file (line 268). `S` will always be `null`.

Do this instead:
```
/obj/machinery/lapvend/proc/process_payment(var/obj/item/weapon/card/id/I, var/obj/item/ID_container)
	var/obj/item/weapon/spacecash/S = null

	if (istype(ID_container, /obj/item/weapon/spacecash))
		S = ID_container

	if (I==ID_container || ID_container == null)
		visible_message("<span class='info'>\The [usr] swipes \the [I] through \the [src].</span>")
	else
		visible_message("<span class='info'>\The [usr] swipes \the [ID_container] through \the [src].</span>")
```

And then carry on with your code.